Framework/DataObjects/src/Workspace2D.cpp

```diff
@@ -134,13 +135,26 @@ size_t Workspace2D::getNumberHistograms() const {
 }
 
 /// get pseudo size
-size_t Workspace2D::size() const { return data.size() * blocksize(); }
+size_t Workspace2D::size() const {
+  return std::accumulate(data.begin(), data.end(), static_cast<size_t>(0),
+                         [](const size_t value, const Histogram1D *histo) {
+                           return value + histo->size();
+                         });
+}
 
 /// get the size of each vector
 size_t Workspace2D::blocksize() const {
-  return (!data.empty())
-             ? static_cast<ISpectrum const *>(data[0])->dataY().size()
-             : 0;
+  if (data.empty()) {
+    return 0;
+  } else {
+    size_t numBins = 0;
+    for (const auto *iter : data)
+      numBins = std::max(numBins, iter->size());
+    if (numBins != data[0]->size())
```

Comment:
Doesn't this still have the same issue as the previous version? What if `data[0]->size()` is the maximum? This will not catch it will it?

Maybe this?
```cpp
for (const auto *iter : data)
  if (iter->size() != data[0]->size())
    throw;
```

And most importantly: Add unit tests for this please!